follow/index.js

```diff
@@ -40,22 +40,19 @@ module.exports = function (fastify, opts, next) {
 
     done()
   })
-
-  next()
 }
 
-function registerRoutes (fastify, opts, done) {
+async function registerRoutes (fastify, opts) {
   const { followService, userClient } = fastify
 
-  fastify.addHook('preHandler', async function (req, reply, done) {
+  fastify.addHook('preHandler', async function (req, reply) {
     try {
       req.user = await userClient.getMe(req)
     } catch (e) {
-      if (!reply.store.config.allowUnlogged) {
-        return done(e)
+      if (!reply.context.config.allowUnlogged) {
```

Comment:
This looks for me like a hack by a fastify professional 💃 I think we can do it better. This config should be passed at `fastify.register`

Reply:
Yeah, it could be a bit strange. And it could not be a good practise.

Anyway the `context` is a configuration per-route not per-plugin.

How could it be written better? *Really*, I appreciate any suggestions.

Reply:
This is actually not so bad, however it's a bit advanced.

Reply:
Yeah, this is a bit advanced. But this is an example built to learn something.

Anyway I'd like to keep this example as simpler as possible. So, is there another way to do this?

Reply:
Go ahead ;)